### ISSUE TITLE
schedulerName field not supported in fieldselector

### ISSUE THREAD

**Issue by astropuffin** [kubernetes/kubernetes#48453]
/kind bug
/sig api-machinery

**What happened**: trying to select pods based on which scheduler they are using

**What you expected to happen**: using fieldSelector would enable me to defer the filtering to the api server instead of using a local for loop over all pods

**How to reproduce it (as minimally and precisely as possible)**:
```
$ curl https://kubernetes.default.svc/api/v1/pods?fieldSelector=spec.schedulerName%3Ddefault -k -H 'Authorization: Bearer REDACTED'
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "field label not supported: spec.schedulerName",
  "reason": "BadRequest",
  "code": 400
}
```

**Anything else we need to know?**:

**Environment**: minikube
```
Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.0", GitCommit:"d3ada0119e776222f11ec7945e6d860061339aad", GitTreeState:"clean", BuildDate:"2017-06-30T09:50:48Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.4", GitCommit:"d6f433224538d4f9ca2f7ae19b252e6fcb66a3ae", GitTreeState:"dirty", BuildDate:"2017-06-22T04:31:09Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
```
In general I would expect fieldSelectors to work for all fields. I understand it's a discouraged feature, but there really isn't any other way to do efficient filtering without just recreating the entire spec in the labels. I found other fields that also don't work, but I am most interested in the newly added schedulerName because I run a custom priority scheduler in my clusters.

**Comment by astropuffin** [comment]
@kubernetes/sig-api-machinery-* because it's purely an API interface issue

**Comment by k82cn** [comment]
That's a reasonable request; we can just get the pods that scheduler interested.

**Comment by mkumatag** [comment]
I have some bandwidth, I can work on this.
/assign

**Comment by lavalamp** [comment]
Not all fields are usable by fieldselector, it is a hand-written list. We want to be careful about adding fields to this list that we have to support in the future (eventually we'll probably have some sort of index). I would ask SIG Architecture for guidance on when fields should be added to the selector list.

BTW, SIG API Machinery doesn't own the *contents* of the API.

**Comment by jaypipes** [comment]
Should this be closed? Looks like the issue has been fixed, yes?

**Comment by dixudx** [comment]
Yes, should close it.

The fix #50582 has already been merged.
